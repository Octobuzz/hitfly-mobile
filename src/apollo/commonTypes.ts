import { Track } from './schemas'

export type HeaderMode = 'light' | 'dark'

export type NavigationState = 'main' | 'profile' | 'auth'

export interface HeaderSettings {
  mode: HeaderMode
  state: NavigationState
}

export type NullableTrack = Track | null
