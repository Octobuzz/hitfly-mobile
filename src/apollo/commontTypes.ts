export type CollectionsType = 'recommended' | 'musicFan'
