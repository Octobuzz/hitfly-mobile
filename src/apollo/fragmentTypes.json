{
  "__schema": {
    "types": [
      {
        "kind": "UNION",
        "name": "AttributeResult",
        "possibleTypes": [
          {
            "name": "User"
          },
          {
            "name": "Track"
          }
        ]
      },
      {
        "kind": "UNION",
        "name": "SearchResult",
        "possibleTypes": [
          {
            "name": "Album"
          },
          {
            "name": "Track"
          },
          {
            "name": "User"
          }
        ]
      },
      {
        "kind": "UNION",
        "name": "CommentResult",
        "possibleTypes": [
          {
            "name": "CommentAlbum"
          },
          {
            "name": "CommentTrack"
          }
        ]
      },
      {
        "kind": "UNION",
        "name": "FavouriteResult",
        "possibleTypes": [
          {
            "name": "Album"
          },
          {
            "name": "Track"
          },
          {
            "name": "Collection"
          },
          {
            "name": "LifehackType"
          }
        ]
      },
      {
        "kind": "UNION",
        "name": "FollowResult",
        "possibleTypes": [
          {
            "name": "FollowUser"
          },
          {
            "name": "FollowMusicGroup"
          }
        ]
      },
      {
        "kind": "UNION",
        "name": "LikeResult",
        "possibleTypes": [
          {
            "name": "LifehackType"
          }
        ]
      },
      {
        "kind": "INTERFACE",
        "name": "UserInterface",
        "possibleTypes": []
      }
    ]
  }
}