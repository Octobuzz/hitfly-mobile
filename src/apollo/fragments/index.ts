export * from './users'
export * from './albums'
export * from './genres'
export * from './tracks'
export * from './collections'
export * from './socialLinks'
