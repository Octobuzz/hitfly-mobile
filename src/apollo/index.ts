import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
