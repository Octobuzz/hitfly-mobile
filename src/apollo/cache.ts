import { InMemoryCache } from 'apollo-cache-inmemory'
import { persistCache } from 'apollo-cache-persist'
import { storageInstance } from 'src/utils'
import gql from 'graphql-tag'

export const defaults = {
  headerSettings: {
    mode: 'dark',
    state: 'main',
    __typename: 'HeaderSettings',
  },
  isPlaying: false,
  activeTrack: null,
  playlist: [],
}

export const typeDefs = gql`
  type Track {
    id: Int!
    title: String
    singer: String
    date: String
    songText: String
    fileUrl: String
    musicWave: [Int]
    length: Int
  }
  type HeaderSettings {
    style: String!
    state: String!
  }
`

export default async (): Promise<InMemoryCache> => {
  const cache = new InMemoryCache({
    freezeResults: true,
  })

  await persistCache({
    cache,
    debug: __DEV__,
    // @ts-ignore
    storage: storageInstance,
  })

  return cache
}
