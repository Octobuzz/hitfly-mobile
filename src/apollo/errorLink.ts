import { Alert } from 'react-native'
import { Operation } from 'apollo-link'
import { GraphQLError } from 'graphql'
import { onError } from 'apollo-link-error'
import { AuthErrorModalRef } from 'src/globalRefs'

const handle401Error = (error: GraphQLError, operation: Operation) => {
  // FIXME: это костыль так как я не понял почему не вытаскивает профиль
  // пришедший после мутации логина
  if (operation.operationName === 'ProfileAvatar') {
    return
  }
  AuthErrorModalRef.show()
}

const handleGraphQLErrors = (
  errors: readonly GraphQLError[],
  operation: Operation,
) => {
  errors.forEach(error => {
    const { message } = error
    // костыль с бэка, по другому ошибку не выявить
    if (message === 'Unauthorized') {
      handle401Error(error, operation)
    }
  })
}

export default onError(({ networkError, graphQLErrors, operation }) => {
  if (networkError) {
    Alert.alert('Ошибка сети', networkError.message)
  }

  if (graphQLErrors) {
    handleGraphQLErrors(graphQLErrors, operation)
  }
})
