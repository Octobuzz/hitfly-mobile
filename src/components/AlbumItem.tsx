import React from 'react'
import { Image } from './Image'
import TextBase from './TextBase'
import { Album, NoAvatarSizeNames } from 'src/apollo'
import { styles } from 'src/constants'
import { useImageSource } from 'src/hooks'
import styled from 'src/styled-components'

const Wrapper = styled.TouchableOpacity`
  width: ${styles.COL2_WIDTH}px;
`

const AlbumImage = styled(Image)`
  height: ${styles.COL2_WIDTH}px;
  width: 100%;
  margin-bottom: 14px;
  border-radius: 4px;
`

const TitleText = styled(TextBase).attrs(() => ({
  numberOfLines: 1,
}))`
  font-family: ${({ theme }) => theme.fonts.medium};
  font-size: 16px;
  line-height: 16px;
  margin-bottom: 8px;
`

const SubTitleText = styled(TextBase).attrs(() => ({
  numberOfLines: 1,
}))`
  font-size: 12px;
  line-height: 12px;
  color: ${({ theme }) => theme.colors.textAlt};
`

interface Props {
  item: Album
  onPress?: (item: Album) => void
}

const AlbumItem: React.FC<Props> = ({ item, onPress }) => {
  const { cover, title, author } = item
  const handlePress = React.useCallback(() => {
    if (onPress) {
      onPress(item)
    }
  }, [onPress, item])

  const source = useImageSource(cover, NoAvatarSizeNames.S_160)

  return (
    <Wrapper onPress={handlePress} testID="albumItem">
      <AlbumImage source={source} />
      <TitleText>{title}</TitleText>
      <SubTitleText>{author}</SubTitleText>
    </Wrapper>
  )
}

export default AlbumItem
