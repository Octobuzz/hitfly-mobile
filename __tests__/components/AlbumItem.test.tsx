import React from 'react'
import {
  render,
  fireEvent,
  ReactTestInstanceExtended,
} from '../../jest/test-utils'
import AlbumItem from 'src/components/AlbumItem'
import { Album } from 'src/apollo'

describe('AlbumItem', () => {
  const item = {
    id: 1,
    title: 'title',
    cover: [{ imageUrl: 'url' }],
    author: 'author',
  } as Album

  it('renders correctly', () => {
    const props = {
      item,
    }
    const { container } = render(<AlbumItem {...props} />)

    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })

  it('can be pressed', () => {
    const props = {
      item,
      onPress: jest.fn(),
    }
    const { getByTestId } = render(<AlbumItem {...props} />)

    fireEvent.press(getByTestId('albumItem'))
    expect(props.onPress).toHaveBeenCalled()
  })
})
