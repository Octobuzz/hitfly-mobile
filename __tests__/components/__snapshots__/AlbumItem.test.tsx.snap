// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumItem renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "width": 351,
      },
    ]
  }
  testID="albumItem"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "borderRadius": 4,
            "height": 351,
            "marginBottom": 14,
            "width": "100%",
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#313131",
          "fontFamily": "GothamPro-Medium",
          "fontSize": 16,
          "lineHeight": 16,
          "marginBottom": 8,
        },
      ]
    }
  >
    title
  </Text>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#606060",
          "fontFamily": "GothamPro",
          "fontSize": 12,
          "lineHeight": 12,
        },
      ]
    }
  >
    author
  </Text>
</TouchableOpacity>
`;
